examples/server_backup/main.go

```diff
@@ -0,0 +1,387 @@
+			options := i.ApplicationCommandData().Options
+			channel := options[0].ChannelValue(s)
+
+			// Acknowledge the command immediately
+			s.InteractionRespond(i.Interaction, &discordgo.InteractionResponse{
+				Type: discordgo.InteractionResponseDeferredChannelMessageWithSource,
+				Data: &discordgo.InteractionResponseData{
+					Content: fmt.Sprintf("Starting backup of channel #%s...", channel.Name),
+				},
+			})
+
+			messages, err := FetchChannelHistory(s, channel.ID)
+			if err != nil {
+				sendFollowUpError(s, i, "Failed to fetch channel history", err)
+				return
+			}
+
+			backup := &BackupData{
+				GuildID:    i.GuildID,
+				GuildName:  channel.Name,
+				BackupTime: time.Now(),
+				Channels: []*ChannelBackup{
+					{
+						ID:       channel.ID,
+						Name:     channel.Name,
+						Type:     channel.Type,
+						Messages: messages,
+					},
+				},
+			}
+
+			// Create backup directory if it doesn't exist
+			if err := os.MkdirAll(*BackupDir, 0755); err != nil {
+				sendFollowUpError(s, i, "Failed to create backup directory", err)
+				return
+			}
+
+			// Save backup to file
+			filename := fmt.Sprintf("%s/%s_%s.json", *BackupDir, channel.Name, time.Now().Format("2006-01-02_15-04-05"))
+			data, err := json.MarshalIndent(backup, "", "  ")
+			if err != nil {
+				sendFollowUpError(s, i, "Failed to marshal backup data", err)
+				return
+			}
+
+			if err := os.WriteFile(filename, data, 0644); err != nil {
+				sendFollowUpError(s, i, "Failed to write backup file", err)
+				return
+			}
+
+			s.InteractionResponseEdit(i.Interaction, &discordgo.WebhookEdit{
+				Content: stringPtr(fmt.Sprintf("✅ Channel backup completed! Backed up %d messages.\nBackup saved as: `%s`",
+					len(messages), filepath.Base(filename))),
+			})
+		},
+		"list-backups": func(s *discordgo.Session, i *discordgo.InteractionCreate) {
+			files, err := os.ReadDir(*BackupDir)
+			if err != nil {
+				s.InteractionRespond(i.Interaction, &discordgo.InteractionResponse{
+					Type: discordgo.InteractionResponseChannelMessageWithSource,
+					Data: &discordgo.InteractionResponseData{
+						Content: "No backups found or backup directory does not exist.",
+					},
+				})
+				return
+			}
+
+			var backupList strings.Builder
+			backupList.WriteString("Available backups:\n```\n")
+			for _, file := range files {
+				if strings.HasSuffix(file.Name(), ".json") {
+					backupList.WriteString(file.Name() + "\n")
+				}
+			}
+			backupList.WriteString("```")
+
+			s.InteractionRespond(i.Interaction, &discordgo.InteractionResponse{
+				Type: discordgo.InteractionResponseChannelMessageWithSource,
+				Data: &discordgo.InteractionResponseData{
+					Content: backupList.String(),
+				},
+			})
+		},
+		"get-backup": func(s *discordgo.Session, i *discordgo.InteractionCreate) {
+			options := i.ApplicationCommandData().Options
+			filename := options[0].StringValue()
+
+			filepath := fmt.Sprintf("%s/%s", *BackupDir, filename)
```

Comment:
that's not very safe... one could ask for `../../../etc/password` etc

also in general this seems quite involved (a lot of code) for an example?